### DISCUSSION TITLE
Is there anything I can do to speed up search and filtering at my end?

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#359] [Help]
My main mindmap, which has been a work in progress since 2017 (more than 5 years) now has more than 42,000 nodes and more than 500,000 words (along with the markup texts for formatting, including images, etc.) It is more than 12MB in size, which is a pretty good size for a text file.

If I open the file in a text editor, I can find any text I want in under a second, as well as all the next instances.

With Freeplane, the same search takes close to a 30 seconds - maybe more.

Is there anything I can do at my end to speed this up? Some kind of index I can create that FP can refer to? It's not intolerable, but there are times, speed would be greatly appreciated.

Thoughts?

~ Bal

**Comment by EdoFro** [comment]
try the **find and replace** command

`Edit->Find->Find and replace… (Ctrl+Shift+F)`

It is way faster.

I think the problem with normal **Search** command for larger maps, is that it has to show you the found node, unfolding all the necessary nodes so it gets visible, and when you jump to the next one, they all get folded again and a new group of nodes gets unfolded.

bye

**Comment by wordmuse** [comment]
Thanks - your answer suggests a feature request... :)
